### PR TITLE
Added integration test

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -30,6 +30,22 @@ dependencies {
     compileOnly "com.google.auto.service:auto-service:1.1.1"
     annotationProcessor "com.google.auto.service:auto-service:1.1.1"
     kapt "com.google.auto.service:auto-service:1.1.1"
+
+    testImplementation 'com.exactpro.th2:junit-jupiter-integration:0.0.1-master-6956603819-5241ee5-SNAPSHOT'
+    testImplementation "org.testcontainers:rabbitmq:1.19.3"
```
Comment (OptimumCode) [inline build.gradle:35]: Why do you need this dependency here?

File: src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+            }
+
+            publishers {
+                pin("pub") {
+                    attributes(TRANSPORT_GROUP_ATTRIBUTE)
+                }
+            }
+        }
+
+    private val resources = ConcurrentLinkedDeque<Pair<String, () -> Unit>>()
```
Comment (OptimumCode) [inline src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt:104]: There is a cleanup extension in th2 junit extension library. Please read the doc: https://github.com/th2-net/junit-jupiter-integration/blob/master/README.md

**Review by OptimumCode — APPROVED** [review]

File: src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+        testFactory.sendMessages(RawMessage.builder().apply {
+            idBuilder().apply {
+                setSessionAlias("test-session-alias")
+                setTimestamp(Instant.now())
+                setDirection(Direction.OUTGOING)
+                setSequence(1)
+            }
+        }.build())
+
+        assertNotNull(eventListener.poll(ofSeconds(2))).also {
```
Comment (OptimumCode) [inline src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt:128]: You should use only one assertion library. Mixing standard junit assertion with Strikt assertion does not look good in my opinion. Especially, considering the fact the Strikt library has everything you need
```suggestion
        expectThat(eventListener.poll(ofSeconds(2))).isNotNull().get { /*...*/ }
```

File: src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+            expectThat(it) {
+                get { eventsList }.apply {
+                    hasSize(1)
+                    withElementAt(0) {
```
Comment (OptimumCode) [inline src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt:132]: You can simplify the assertion here
```suggestion
                get { eventsList }.single()
```

**Review by OptimumCode — APPROVED** [review]

File: src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt (hunk 3 above)
Comment (Nikita-Smirnov-Exactpro) [reply src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt:128]: expectThat(...) expect lambda as the second argument

Comment (OptimumCode) [reply src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt:128]: It has an overload that has only one parameter. Please, check the API one more time

Comment (Nikita-Smirnov-Exactpro) [reply src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt]: I found, thanks

File: src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt [hunk 5]
```diff
@@ -125,24 +126,22 @@ class ApplicationIntegrationTest {
...
-                        get { scope }.isEqualTo(rootEventId.scope)
-                    }
-                    get { parentId }.isEqualTo(rootEventId)
-                    get { attachedMessageIdsList }.isEmpty()
-                    get { body.toString(Charsets.UTF_8) }.isEqualTo(
-                        """
-                            [{"data":"java.net.ConnectException: Connection refused (Connection refused)","type":"message"}]
-                        """.trimIndent()
-                    )
+        expectThat(eventListener.poll(ofSeconds(2))).isNotNull().apply {
```
Comment (OptimumCode) [inline src/test/kotlin/com/exactpro/th2/http/client/ApplicationIntegrationTest.kt:129]: Just a suggestion:

The Strikt API uses the builder pattern. This means you can chain method calls to do the assertion (without `apply` function).
For example,
```kotlin
expectThat(eventListener.poll(ofSeconds(2)))
 .isNotNull()
 .get { eventsList }
 .single().and {
    // other assertion
  }
```
In my opinion, it looks much clearer than the current approach

**Review by OptimumCode — APPROVED** [review]

**Review by OptimumCode — APPROVED** [review]

**Pull request merged**